react/ContactsListModal/AddContactButton.jsx

```diff
@@ -1,22 +1,50 @@
-import React from 'react'
-import { withClient } from 'cozy-client'
+import React, { useEffect } from 'react'
+import { withClient, models, queryConnect } from 'cozy-client'
+import CozyRealtime from 'cozy-realtime'
 import AppLinker, { generateWebLink } from '../AppLinker'
 import { ButtonLink } from '../Button'
+import compose from 'lodash/flowRight'
 
 const DumbAddContactButton = props => {
-  const { client, ...rest } = props
-
-  const cozyURL = new URL(client.getStackClient().uri)
-  const { cozySubdomainType } = client.getInstanceOptions()
-  const contactsAppSlug = 'contacts'
-  const contactsAppHref = generateWebLink({
-    cozyUrl: cozyURL.origin,
-    slug: contactsAppSlug,
-    subDomainType: cozySubdomainType
+  const { client, apps, ...rest } = props
+
+  const wantedApp = { slug: 'contacts' }
+  const installedApp = models.applications.isInstalled(apps.data, {
+    slug: 'contacts'
   })
 
+  let href
+
+  if (installedApp) {
+    const cozyURL = new URL(client.getStackClient().uri)
+    const { cozySubdomainType } = client.getInstanceOptions()
+    href = generateWebLink({
+      cozyUrl: cozyURL.origin,
+      slug: installedApp.attributes.slug,
+      subDomainType: cozySubdomainType
+    })
+  } else {
+    href = models.applications.getStoreInstallationURL(apps.data, wantedApp)
+  }
+
+  useEffect(() => {
+    const realtime = new CozyRealtime({ client })
+    const refreshApps = () => {
+      apps.fetch()
+    }
+
+    realtime.subscribe('created', 'io.cozy.apps', refreshApps)
```

Comment:
We should try / await / catch this subscribe and put a warning to tell the dev that to use this feature the app should have the rights on `io.cozy.apps`

Reply:
:+1: